src/main/java/uk/gov/hmcts/ccd/endpoint/ui/QueryEndpoint.java

```diff
@@ -214,4 +200,23 @@ public CaseEventTrigger getEventTriggerForCase(@PathVariable("uid") String userI
                                                        eventTriggerId,
                                                        ignoreWarning);
     }
+
+    @Transactional
+    @RequestMapping(value = "/caseworkers/{uid}/jurisdictions/{jid}/case-types/{ctid}/cases/{cid}/events/{eventId}",
+        method = RequestMethod.GET)
+    @ApiOperation(value = "Fetch case history for the event")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Displayable case data")
+    })
+    public CaseHistoryView getCaseHistoryForEvent(@PathVariable("jid") final String jurisdictionId,
+                                                  @PathVariable("ctid") final String caseTypeId,
+                                                  @PathVariable("cid") final String caseReference,
+                                                  @PathVariable("eventId") final Long eventId) {
+        Instant start = Instant.now();
+        CaseHistoryView caseView = getCaseViewOperation.execute(jurisdictionId, caseTypeId, caseReference, eventId);
```

Comment:
can we rename caseView to caseHistoryView?

Reply:
sure

Reply:
done